pkg/network/tracer/chain_conntracker.go

```diff
@@ -0,0 +1,106 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+// Package tracer implements the functionality of the network tracer
+package tracer
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/DataDog/datadog-agent/pkg/network"
+	"github.com/DataDog/datadog-agent/pkg/network/netlink"
+)
+
+type chainConntracker struct {
+	ctks []netlink.Conntracker
+}
+
+func chainConntrackers(ctks ...netlink.Conntracker) netlink.Conntracker {
+	if len(ctks) == 1 {
+		return ctks[0]
+	}
+
+	// filter out no-ops
+	filtered := ctks[:0]
+	for _, ctk := range ctks {
+		if ctk != nil && ctk.GetType() != "" {
+			filtered = append(filtered, ctk)
+		}
+	}
+
+	if len(filtered) == 0 {
+		return netlink.NewNoOpConntracker()
+	}
+
+	if len(filtered) == 1 {
+		return filtered[0]
+	}
+
+	return &chainConntracker{
+		ctks: filtered,
+	}
+}
+
+// Describe returns all descriptions of the collector
+func (ct *chainConntracker) Describe(descs chan<- *prometheus.Desc) {
+	for _, ctk := range ct.ctks {
+		ctk.Describe(descs)
+	}
+}
+
+// Collect returns the current state of all metrics of the collector
+func (ct *chainConntracker) Collect(metrics chan<- prometheus.Metric) {
+	for _, ctk := range ct.ctks {
+		ctk.Collect(metrics)
+	}
+}
+
+func (ct *chainConntracker) GetTranslationForConn(c *network.ConnectionTuple) *network.IPTranslation {
+	for _, ctk := range ct.ctks {
+		if trans := ctk.GetTranslationForConn(c); trans != nil {
+			return trans
+		}
+	}
+
+	return nil
+}
+
+// GetType returns a string describing whether the conntracker is "ebpf" or "netlink"
+func (ct *chainConntracker) GetType() string {
+	return "chain"
+}
+
+func (ct *chainConntracker) DeleteTranslation(c *network.ConnectionTuple) {
```

Comment:
nit: exported methods should be have comments (why doesn't the linter care?)

Reply:
:shrug: its the same for other conntrackers too. I'll add some comments anyway.